src/lib.ts

```diff
@@ -0,0 +1,180 @@
+export const CONFIG_PATH = xdgAppPaths("webstudio").config();
+export const CONFIG_FILE = path.join(CONFIG_PATH, "config.json");
+export let pm: string | null = null;
+export const supportedBuildTypes = ['remix-app-server', 'express', 'architect', 'flyio', 'netlify', 'vercel', 'cloudflare-pages', 'cloudflare-workers', 'deno'];
+
+const HELP = `Usage:
+    $ webstudio commands [flags...]
+  Commands:
+    login <shared link>             Login to Webstudio with shared link
+    sync <projectId>                Download a project's site data
+    build [projectId]               Build a site
+  Flags:
+    --type, -t                      Build type chosen during build command (default: remix-app-server) 
+                                    (options: ${supportedBuildTypes.join(', ')})
+    --debug                         Enable debug mode
+    --help, -h                      Show this help message
+    --version, -v                   Show the version of this script
+`;
+export const showHelp = () => {
+    console.log(HELP);
+}
+export const detectPackageManager = async () => {
+    const originalverbose = $.verbose;
+    $.verbose = false;
+    try {
+        const pnpm = await $`pnpm --version`.exitCode;
+        const yarn = await $`yarn --version`.exitCode;
+        const npm = await $`npm --version`.exitCode;
+        if (pnpm === 0) {
+            return pm = 'pnpm';
+        }
+        if (yarn === 0) {
+            return pm = 'yarn';
+        }
+        if (npm === 0) {
+            return pm = 'npm';
+        }
+        throw new Error('No package manager found');
+    } catch (e) {
+        throw new Error(e);
+    } finally {
+        $.verbose = originalverbose;
+    }
+
+    $.verbose = originalverbose;
+}
+export const prepareConfigPath = async () => {
+    await $`mkdir -p ${CONFIG_PATH}`
+    await checkConfig();
+}
+export const prepareBuildDir = async () => {
+    await $`rm -rf ${BUILD_DIR}`;
+    await $`mkdir -p ${BUILD_DIR}/app/__generated__`;
+    await $`mkdir -p ${BUILD_DIR}/app/routes`;
+}
+export const checkSiteData = async (args) => {
+    const projectId = args.positionals[1];
+    return await $`ls -1 app/${projectId}.json > /dev/null 2>&1`.exitCode;
+}
+
+export const prepareDefaultRemixConfig = async (type: string) => {
+    console.log(`Preparing default configurations for ${type}...`)
+    await $`cp ./templates/defaults/root.tsx ./${BUILD_DIR}/app`
+    const def = await fs.readFile('./templates/defaults/package.json', 'utf-8');
+    const defaultJson = JSON.parse(def);
+    const template = await fs.readFile(`./templates/${type}/package.json`, 'utf-8');
+    const templateJson = JSON.parse(template);
+    const merged = deepmerge(defaultJson, templateJson);
+    await fs.writeFile(`./${BUILD_DIR}/package.json`, JSON.stringify(merged, null, 2));
+
+    await $`cp ./templates/defaults/template.tsx ./${BUILD_DIR}`
+    await $`cp ./templates/defaults/remix.config.js ./${BUILD_DIR}`
+
+    switch (type) {
```

Comment:
where is this particular list coming from? is this what remix's cli supports?

Reply:
can this list of copied files be simplied by a convetion of having a folder where all files get copied?

Reply:
this list could be completely gone I think

Reply:
> where is this particular list coming from? is this what remix's cli supports?

Yes, it is coming from Remix's CLI. 



> can this list of copied files be simplied by a convetion of having a folder where all files get copied?

I think it is already simple. Everything is copied from template + package.json is merged with our custom stuffs.